
\renewcommand{\hbAppendixPrefix}{C}

\renewcommand{\thefigure}{\hbAppendixPrefix\arabic{figure}}
\setcounter{figure}{0}
\renewcommand{\thetable}{\hbAppendixPrefix\arabic{table}}
\setcounter{table}{0}
\renewcommand{\theequation}{\hbAppendixPrefix\arabic{equation}}
\setcounter{equation}{0}

\section{Supplementary Material 1 for Chapter \ref{ch:ctherm}} \label{apx:sm1-ctherm}

%\documentclass[dvipsnames]{article}
%\usepackage[sm]{paper}
%%\usepackage{longtable}
%\usepackage{supertabular}
%\addbibresource{bibliography.bib}

% Command to easily format gene knockouts
%\makeatletter
%\newcommand\ko[1]{{\@for\tmp:=#1\do{\textit{$\Delta$\tmp}}}}



%\title{ \textbf{Supplementary Material 1:} \\Development of an updated genome-scale metabolic model of \textit{Clostridium thermocellum} and its application for integration of multi-omics datasets}
%\author[1,2]{Sergio Garcia}
%\author[2,3]{R. Adam Thompson}
%\author[2,5]{Richard J. Giannone}
%\author[2,4]{Satyakam Dash}
%\author[2,4]{Costas D. Maranas}
%\author[1,2,3,*]{Cong T. Trinh}
%\affil[1]{Department of Chemical and Biomolecular Engineering, The University of Tennessee, Knoxville, TN, United States}
%\affil[2]{Center for Bioenergy Innovation, Oak Ridge National Laboratory Oak Ridge, TN, United States}
%\affil[3]{Bredesen Center for Interdisciplinary Research and Graduate Education, The University of Tennessee, Knoxville and Oak Ridge National Laboratory, Oak Ridge, TN, USA}
%\affil[4]{Department of Chemical Engineering, The Pennsylvania State University, University Park, PA, United States}
%\affil[5]{Chemical Sciences Division, Oak Ridge National Laboratory. Oak Ridge, TN, United States.}
%\affil[*]{Corresponding author: 1512 Middle Dr, DO432, Deparment of Chemical and Biomolecular Engineering, University of Tennesse, Knoxville, TN 37996, United States. Tel: 865-974-2181. Email: ctrinh@utk.edu.}
%
%
%\begin{document}
%\maketitle
%%\tableofcontents
%%\listoftables
%\newpage

%\section*{Consistent cases}

%\todo[inline]{A list of reaction abreviations to names can be added at the end but otherwise it is too much for one table, also consider adding the model in "table form" for easy reference}

%\todo[inline]{Add continuation caption}
% A way to add continuation caption:
%https://tex.stackexchange.com/questions/11380/how-to-repeat-top-rows-column-headings-on-every-page
% --- See longtable instead
% https://tex.stackexchange.com/questions/26462/make-a-table-span-multiple-pages


%\begin{table}[!ht]
%\caption[Consistent reactions between flux simulations and proteomic data]{The 70 consistent reactions in the
%\textit{$\Delta$hydG-$\Delta$ech} case study and their associated fold changes. The biomass reaction is not included due to size. This table is continued in the following pages.}

%\topcaption{The 70 consistent reactions in the \protect\ko{hydG,ech} case study and their associated fold changes. The biomass reaction is not included due to size. This table is continued in the following pages.}%

%\begin{longtable}{ccc}
%\caption {Some Members of the Suggested Estimator}\label{Tab:1}\\\toprule
%\endfirsthead
%\caption* {\textbf{Table \ref{Tab:1} Continued:} Blablabla}\\\toprule
%\endhead
%\endfoot
%\bottomrule
%\endlastfoot
%A subset of $\overline{y}_{NH}$ & $\textbf{a}$ & $\textbf{b}$\\
%\midrule
%test1 & 1 & 0 \\
%\midrule
%test2 & 1 & 1 \\
%\midrule
%test3 & \newline 1 & \newline $\beta_{2} \left(x\right)$ \\
%\midrule
%A subset of $\overline{y}_{NH}$ & $\textbf{a}$ & $\textbf{b}$\\
%\midrule
%test4 & $C_{x} $ & $\rho_{yx} $ \\
%\midrule
%test5 & $\rho _{yx} $ & $C_{x} $ \\
%\midrule
%test6 & $\rho _{yx} $ \\
%A subset of $\overline{y}_{NH}$ & $\textbf{a}$ & $\textbf{b}$\\
%\midrule
%test1 & 1 & 0 \\
%\midrule
%test2 & 1 & 1 \\
%\midrule
%test3 & \newline 1 & \newline $\beta_{2} \left(x\right)$ \\
%\midrule
%A subset of $\overline{y}_{NH}$ & $\textbf{a}$ & $\textbf{b}$\\
%\midrule
%test4 & $C_{x} $ & $\rho_{yx} $ \\
%\midrule
%test5 & $\rho _{yx} $ & $C_{x} $ \\
%\midrule
%test6 & $\rho _{yx} $ \\
%\end{longtable}

\newpage
\input{./figures/chapter-6/top_up.tex}
\newpage
%\end{table}

% Supertabular
%\topcaption[Consistent reactions between flux simulations and proteomic data]{The 70 consistent reactions in the
%\textit{$\Delta$hydG-$\Delta$ech} case study and their associated fold changes. The biomass reaction is not included due to size. This table is continued in the following pages.}
%\input{./figures/chapter-6/top_up.tex}


%\pagebreak
%\section{Flux simulations to further elucidate possible fate of \textit{nadph}}

\begin{table}[ht]
\caption[Simulated fluxes for \textit{$\Delta$hydG-$\Delta$ech}]{pFBA simulated fluxes from \textit{$\Delta$hydG-$\Delta$ech} case study. Only includes reactions involving NADPH or exchange reactions that have different flux between wild-type and mutant. The biomass reaction is not included due to size. Additionally, we also excluded reactions with the same fold change magnitude as the biomass reaction ($|\textrm{FC}|=0.26$), because their fluxes are likely fully correlated.}
\input{./figures/chapter-6/nadph_and_ex.tex}
\end{table}

%\thispagestyle{empty}
\begin{table}[h]
    \caption{Reaction deletions sorted by appeareance frequency (counts) in the designs of the Pareto front for $\alpha=6,\beta=0$.}
    \centering
    \input{./figures/chapter-6/table-deletions.tex}
    \label{tab:deletions}
\end{table}

